addons/website_slides/models/slide_channel.py

```diff
@@ -327,24 +414,47 @@ def _get_default_enroll_msg(self):
     can_comment = fields.Boolean('Can Comment', compute='_compute_action_rights', compute_sudo=False)
     can_vote = fields.Boolean('Can Vote', compute='_compute_action_rights', compute_sudo=False)
 
+    def _compute_allow_sharing_preview(self):
+        for record in self:
+            record.allow_sharing_preview = True
+
+    @api.depends('channel_partner_all_ids', 'channel_partner_all_ids.member_status')
+    def _compute_partner_ids(self):
+        for record in self:
+            enrolled_partner_ids = record.channel_partner_ids.partner_id.ids or []
+            record.partner_ids = record.partner_all_ids.filtered(lambda partner: partner.id in enrolled_partner_ids)
+
+    def _search_partner_ids(self, operator, value):
+        if isinstance(value, int) and operator == 'in':
+            value = [value]
+        enrolled_channel_partner_ids = self.env['slide.channel.partner'].sudo().search([
+            ('member_status', '!=', 'invited'),
+            ('partner_id', operator, value),
+        ])
+        return [('id', 'in', enrolled_channel_partner_ids.channel_id.ids)]
+
     @api.depends('slide_ids.is_published')
     def _compute_slide_last_update(self):
         for record in self:
             record.slide_last_update = fields.Date.today()
 
-    @api.depends('channel_partner_ids.channel_id')
-    def _compute_members_count(self):
-        read_group_res = self.env['slide.channel.partner'].sudo()._read_group([('channel_id', 'in', self.ids)], ['channel_id'], 'channel_id')
-        data = dict((res['channel_id'][0], res['channel_id_count']) for res in read_group_res)
-        for channel in self:
-            channel.members_count = data.get(channel.id, 0)
-
-    @api.depends('channel_partner_ids.channel_id', 'channel_partner_ids.completed')
-    def _compute_members_done_count(self):
-        read_group_res = self.env['slide.channel.partner'].sudo()._read_group(['&', ('channel_id', 'in', self.ids), ('completed', '=', True)], ['channel_id'], 'channel_id')
-        data = dict((res['channel_id'][0], res['channel_id_count']) for res in read_group_res)
+    @api.depends('channel_partner_all_ids.channel_id', 'channel_partner_all_ids.member_status')
+    def _compute_members_counts(self):
+        read_group_res = self.env['slide.channel.partner'].sudo()._read_group(
+            [('channel_id', 'in', self.ids)],
+            ['channel_id'],
+            ['channel_id', 'member_status'],
+            lazy=False)
+        data_invited = {res['channel_id'][0]: res['__count'] for res in read_group_res if res.get('member_status') == 'invited'}
+        data_joined = {res['channel_id'][0]: res['__count'] for res in read_group_res if res.get('member_status') == 'joined'}
+        data_ongoing = {res['channel_id'][0]: res['__count'] for res in read_group_res if res.get('member_status') == 'ongoing'}
+        data_completed = {res['channel_id'][0]: res['__count'] for res in read_group_res if res.get('member_status') == 'completed'}
```

Comment:
You could avoid looping 4 times over the same data with something like
```suggestion
        members_status_counts_per_channel = {status: {} for status in ["invited", "joined", "ongoing", "completed"]}

        for res in read_group_res:
            members_status_counts_per_channel[res['member_status']][res['channel_id'][0]] = res['__count']
```
(And if `member_status` can be NULL, add a constraint in the domain excluding those)